lib/Test2/Plugin/DataLine.pm

```diff
@@ -0,0 +1,117 @@
+package Test2::Plugin::DataLine;
+use strict;
+use warnings;
+
+use Test2::Event::V2;
+
+our $VERSION = '0.000154';
+
+use B();
+
+use Test2::API qw{
+    test2_add_callback_post_load
+    test2_stack
+};
+
+sub import {
+    my $class = shift;
+
+    test2_add_callback_post_load(sub {
+        my $hub = test2_stack()->top;
+
+        $hub->pre_filter(
+            sub {
+                my ($hub, $event) = @_;
+
+                if ($event->causes_fail) {
+                    my $fd = $event->facet_data;
+
+                    if (my $line = $.) {
+                        my $fh = eval '$${^LAST_FH}' || do { # Added in 5.18, the do is fallback
+                            my $out;
+                            local $SIG{__WARN__} = sub {
+                                my $msg = shift;
+                                $out = $msg;
+                            };
+                            warn "blah";
+                            $out =~ m/<(.+)> line $line/ ? $1 : '?';
```

Comment:
Could we make this regex more specific? What if the test name contains ` line `?

I don’t understand why there are `<>` in this regex. You don’t want to utilize `//x` for compatibility, right?